### ISSUE TITLE
Wrong operation in Artificial Bee Colony. 

### ISSUE THREAD

**Issue by peacemo** [NiaOrg/NiaPy#600]
**Steps to reproduce**
1. Debug a task for  Artificial Bee Colony. 
2. Add breakpoints where the `trials` parameters could be observed. 
3. The individuls with `trial` which are large_equel(>=) to `limit` are not updated thoroghly. 

**Expected behavior**
All the individuals whoes `trial` >= limit should updates. 

**Actual behavior**
Only the individual with largest `trail` is compared to `limit` and updated. The other individuals with `trial` > `limit` are skipped. 

These are the codes accorddingly:

https://github.com/NiaOrg/NiaPy/blob/0c50a77caca3c2c9128c7be8abf958fa6967a9e2/niapy/algorithms/basic/abc.py#L210C9-L214C81

```python
mi = np.argmax(trials)
if trials[mi] >= self.limit:
    population[mi], trials[mi] = SolutionABC(task=task, rng=self.rng), 0
    if population[mi].f < best_fitness:
        best_x, best_fitness = population[mi].x.copy(), population[mi].f
```

**System configuration**

 - OS: Windows 11
 - Python version: 3.12

--------
With the 'limit' set to 10

**Trials before update**

![125c7ee5f1702270ed0d06895e51568](https://github.com/user-attachments/assets/f0434645-3f14-4c9f-aa24-72d88bbb49a6)

There were 2 individuals should be updated. 

**Trials after update**

![08538516c0e053e1c3da6fd5a128e1f](https://github.com/user-attachments/assets/793ea9ae-3eb1-4009-8a47-9659fbbf6d42)

Only the one with biggest `trial` was updated. 

In some extreme conditions, the update proccedure would get stuck which means the algorithm may not get out of the local optimal.

**Comment by firefly-cpp** [comment]
Resolved in pull request #601. @peacemo, a new version of the framework has recently been released. Please test the changes of your proposed fix in the latest release.

**Comment by peacemo** [comment]
> Resolved in pull request #601. @peacemo, a new version of the framework has recently been released. Please test the changes of your proposed fix in the latest release.

I noticed that the PR has already been merged. Is there anything I can do for this issue or it can be closed.

**Comment by peacemo** [comment]
I'll  close the issue, if there are something else associated with this issue, please let me know!
